src/axom/quest/IntersectionShaper.hpp

```diff
@@ -364,41 +364,99 @@ class IntersectionShaper : public Shaper
     loadShapeInternal(shape, m_percentError, m_revolvedVolume);
 
     // Filter the mesh, store in m_surfaceMesh.
-    SegmentMesh* newm =
-      filterMesh(dynamic_cast<const SegmentMesh*>(m_surfaceMesh));
-    delete m_surfaceMesh;
-    m_surfaceMesh = newm;
+    if(shape.getGeometry().getFormat() == "c2c")
+    {
+      SegmentMesh* newm =
+        filterMesh(dynamic_cast<const SegmentMesh*>(m_surfaceMesh));
+      delete m_surfaceMesh;
+      m_surfaceMesh = newm;
+    }
   }
 
 public:
-//@{
-//!  @name Functions related to the stages for a given shape
+  //@{
+  //!  @name Functions related to the stages for a given shape
 
-/// Initializes the spatial index for shaping
 #if defined(AXOM_USE_RAJA) && defined(AXOM_USE_UMPIRE)
+
+  // Prepares the ProE mesh cells for the spatial index
   template <typename ExecSpace>
-  void prepareShapeQueryImpl(klee::Dimensions shapeDimension,
-                             const klee::Shape& shape)
+  void prepareProECells()
   {
-    SLIC_INFO(axom::fmt::format(
-      "{:-^80}",
-      axom::fmt::format(
-        "Running intersection-based shaper in execution Space: {}",
-        axom::execution_space<ExecSpace>::name())));
+    // Number of tets in mesh
+    m_tetcount = m_surfaceMesh->getNumberOfCells();
 
-    // Save current/default allocator
-    const int current_allocator = axom::getDefaultAllocatorID();
+    m_tets = axom::allocate<TetrahedronType>(m_tetcount);
+    axom::StackArray<IndexType, 1> tets_dimensions = {m_tetcount};
+    axom::ArrayView<TetrahedronType> tets_view(m_tets, tets_dimensions);
 
-    // Determine new allocator (for CUDA/HIP policy, set to Unified)
-    // Set new default to device
-    axom::setDefaultAllocator(axom::execution_space<ExecSpace>::allocatorID());
+    // Initialize tetrahedra
+    axom::Array<IndexType> nodeIds(4);
+    axom::Array<Point3D> pts(4);
 
-    const auto& shapeName = shape.getName();
-    AXOM_UNUSED_VAR(shapeDimension);
-    AXOM_UNUSED_VAR(shapeName);
+    for(int i = 0; i < m_tetcount; i++)
+    {
+      m_surfaceMesh->getCellNodeIDs(i, nodeIds.data());
 
-    SLIC_INFO(axom::fmt::format("Current shape is {}", shapeName));
+      m_surfaceMesh->getNode(nodeIds[0], pts[0].data());
+      m_surfaceMesh->getNode(nodeIds[1], pts[1].data());
+      m_surfaceMesh->getNode(nodeIds[2], pts[2].data());
+      m_surfaceMesh->getNode(nodeIds[3], pts[3].data());
 
+      m_tets[i] = TetrahedronType({pts[0], pts[1], pts[2], pts[3]});
+    }
```

Comment:
> Is there a way to do this w/in a kernel in the ExecSpace too?

I did not explore it in this PR, as a lot of these calls are to `mint::UnstructuredMesh` functions which are not host/device annotated yet. 
The functions also override virtual `mint::Mesh` functions, so that might complicate things.